spyder/widgets/ipythonconsole/shell.py

```diff
@@ -243,11 +243,11 @@ def handle_exec_method(self, msg):
                         self.sig_namespace_view.emit(view)
                     else:
                         view = None
-                elif 'get_var_properties' in method:
+                elif 'get_var_properties' in method and 'text/plain' in data:
```

Comment:
I think this is not right. It's clearer if you write it as:

```python
elif 'get_var_properties' in method:
    if data is not None and 'text/plain' in data:
        properties = ast.literal_eval(data['text/plain'])
        self.sig_var_properties.emit(properties)
```

At least it has the same form as the other `if` clauses here, which makes things easier to read.